src/Lib/Util/JWT.hs

```diff
@@ -56,9 +56,8 @@ decodeAndVerifyJWTToken token = do
   secret <- JWT.secret <$> asks jwtSecret
   timeNow <- JWT.numericDate <$> liftIO getPOSIXTime
   pure $ JWT.claims <$> JWT.decodeAndVerifySignature secret token >>= \claimsSet ->
-    case (,) <$> timeNow <*> JWT.exp claimsSet of
-      Just (now, expiryTimeStatedInToken) ->
-        if expiryTimeStatedInToken < now then
-          Nothing
-        else
-          jwtPayloadFromMap $ JWT.unregisteredClaims claimsSet
+    (,) <$> timeNow <*> JWT.exp claimsSet >>= \(now, expiryTimeStatedInToken) ->
+      if expiryTimeStatedInToken < now then
+        Nothing
+      else
+        jwtPayloadFromMap $ JWT.unregisteredClaims claimsSet
```

Comment:
I think it's better to rewrite this code using `do`-notation for `Maybe` monad to make it cleaner. Probably putting it under `let` first in order to be able to specify type of this action.